lib/IR/IRUtils.cpp

```diff
@@ -51,3 +51,43 @@ const Value *glow::getOrigin(const Value *V) {
   }
   return V;
 }
+
+size_t glow::getOriginOffset(Value *V) {
+  size_t off = 0;
+
+  // Since each TensorView can either maintain or decrease the size of a buffer,
+  // the linearized offset of a TensorView into its underlying buffer can be
+  // computed by adding together the linearized offsets of intermediate
+  // TensorViews relative to their sources.
+  while (true) {
+    auto *TVI = dyn_cast<TensorViewInst>(V);
+    if (!TVI) {
+      return off;
+    }
+
+    llvm::ArrayRef<size_t> offsets = TVI->getOffsets();
+    llvm::ArrayRef<size_t> srcDims = TVI->getSrc()->getType()->dims();
+
+    size_t numOffsets = offsets.size(), numSrcDims = srcDims.size();
+    size_t sliceSize = 1;
```

Comment:
`Type` has a `getSliceSize` method. May be it could be useful here to shorten the code?

Reply:
Ah I still need to do this, do not accept yet.